File: src/Entity/Club.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping\InverseJoinColumn;
use App\Repository\ClubRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Doctrine\ORM\Mapping\JoinColumn;
use Doctrine\ORM\Mapping\JoinTable;

#[ORM\Entity(repositoryClass: ClubRepository::class)]
class Club
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    #[Groups(['read.Pays', 'read.Club'])]
    private $id;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(['read.Club', 'persist.Club'])]
    private $nom;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(['read.Club', 'persist.Club'])]
    private $adresse;

    #[ORM\ManyToOne(targetEntity: Pays::class)]
    #[ORM\JoinColumn(referencedColumnName: 'code')]
    #[Groups(['read.Club', 'persist.Club'])]
    private $pays;

    #[ORM\Column(type: 'date', nullable: true)]
    #[Groups(['read.Club', 'persist.Club'])]
    private $dateCreation;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    #[Groups(['read.Club', 'persist.Club'])]
    private $siteWeb;

    #[ORM\OneToMany(targetEntity: LogoClub::class, mappedBy: 'club', cascade: ['persist', 'remove'], fetch: 'EAGER')]
    #[Groups(['read.Club', 'persist.Club'])]
    private $logos;

    #[ORM\Column(type: 'integer', nullable: true)]
    #[Groups(['read.Club', 'persist.Club'])]
    private $idTransfermarkt;

    #[ORM\ManyToOne(targetEntity: Stade::class, cascade: ['persist', 'remove'], fetch: 'EAGER')]
    #[Groups(['read.Club', 'persist.Club'])]
    private $stade;

    #[JoinTable(name: 'club_couleur_club')]
    #[JoinColumn(name: 'club_id', referencedColumnName: 'id')]
    #[InverseJoinColumn(name: 'couleur_club_id', referencedColumnName: 'id')]
    #[ORM\ManyToMany(targetEntity: CouleurClub::class, cascade: ['persist'])]
    #[Groups(['read.Club', 'persist.Club'])]
    private $couleurs;

    public function __construct()
    {
        $this->logos = new ArrayCollection();
        $this->couleurs = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }
    
    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(?string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getPays(): ?Pays
    {
        return $this->pays;
    }

    public function setPays(?Pays $pays): self
    {
        $this->pays = $pays;

        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(?\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;

        return $this;
    }

    public function getSiteWeb(): ?string
    {
        return $this->siteWeb;
    }

    public function setSiteWeb(?string $siteWeb): self
    {
        $this->siteWeb = $siteWeb;

        return $this;
    }

    /**
     * @return Collection<int, LogoClub>
     */
    public function getLogos(): Collection
    {
        return $this->logos;
    }

    public function addLogo(LogoClub $logo): self
    {
        if (!$this->logos->contains($logo)) {
            $this->logos[] = $logo;
            $logo->setClub($this);
        }

        return $this;
    }

    public function removeLogo(LogoClub $logo): self
    {
        if ($this->logos->removeElement($logo)) {
            // set the owning side to null (unless already changed)
            if ($logo->getClub() === $this) {
                $logo->setClub(null);
            }
        }

        return $this;
    }

    public function getIdTransfermarkt(): ?int
    {
        return $this->idTransfermarkt;
    }

    public function setIdTransfermarkt(?int $idTransfermarkt): self
    {
        $this->idTransfermarkt = $idTransfermarkt;

        return $this;
    }

    public function getStade(): ?Stade
    {
        return $this->stade;
    }

    public function setStade(?Stade $stade): self
    {
        $this->stade = $stade;

        return $this;
    }

    /**
     * @return Collection<int, CouleurClub>
     */
    public function getCouleurs(): Collection
    {
        return $this->couleurs;
    }

    public function addCouleur(CouleurClub $couleur): self
    {
        if (!$this->couleurs->contains($couleur)) {
            $this->couleurs[] = $couleur;
        }

        return $this;
    }

    public function removeCouleur(CouleurClub $couleur): self
    {
        $this->couleurs->removeElement($couleur);

        return $this;
    }
}

File: public/scrapp/lib/Classes/Club.py
from lib.Classes.Pays import Pays
from lib.Classes.Stade import Stade
from lib.Classes.CouleurClub import CouleurClub
from lib.Classes.LogoClub import LogoClub


import json as oejson

class Club:

    def __new__(cls):
        return super(Club, cls).__new__(cls)

    def __init__(self):
        self.uri = None
        self.id = None
        self.nom = None
        self.adresse = None
        self.pays = Pays()
        self.dateCreation = None
        self.siteWeb = None
        self.couleurs = []
        self.logos = []
        self.idTransferMarkt = None
        self.stade = Stade()

    def toJson(self, schema: str = "") -> dict:
        if schema == 'persist.Joueur':
            json = {}
        elif schema == 'persist.Club':
            json = {
                "nom": self.nom,
                "adresse": self.adresse,
                "pays": f"/api/pays/{self.pays.code}" if self.pays.code is not None else None,
                "dateCreation": self.dateCreation,
                "siteWeb": self.siteWeb,
                "couleurs": [couleur.toJson(schema=schema) for couleur in self.couleurs],
                "logos": [logo.toJson(schema=schema) for logo in self.logos],
                "idTransfermarkt": int(self.idTransferMarkt) if self.idTransferMarkt is not None else None,
                "stade": f"/api/stade/{self.stade.id}" if self.stade.id is not None else None
            }
        else:
            json = {
                "id": self.id,
                "nom": self.nom,
                "adresse": self.adresse,
                "pays": self.pays.toJson(schema=schema),
                "dateCreation": self.dateCreation,
                "siteWeb": self.siteWeb,
                "couleurs": [couleur.toJson(schema=schema) for couleur in self.couleurs],
                "logos": [logo.toJson(schema=schema) for logo in self.logos],
                "idTransfermarkt": self.idTransferMarkt,
                "stade": self.stade.toJson(schema=schema) 
            }

        return json


    def fromJson(self, json: dict):
        if isinstance(json, str) :
            self.uri = json

            return self
        
        self.uri = json['@id'] if json.get('@id') else self.uri
        self.id = json['id'] if json.get('id') else self.id
        self.nom = json['nom'] if json.get('nom') else self.nom
        self.adresse = json['adresse'] if json.get('adresse') else self.adresse
        self.pays = Pays().fromJson(
            json=json['pays']) if json.get('pays') else self.pays
        self.dateCreation = json['dateCreation'] if json.get(
            'dateCreation') else self.dateCreation
        self.siteWeb = json['siteWeb'] if json.get(
            'siteWeb') else self.siteWeb
        self.couleurs = [CouleurClub().fromJson(json=couleur)
                         for couleur in json['couleurs']] if json.get('couleurs') else self.couleurs
        self.logos = [LogoClub().fromJson(json=logo)
                      for logo in json['logos']] if json.get('logos') else self.logos
        self.idTransferMarkt = json['idTransfermarkt'] if json.get(
            'idTransfermarkt') else self.idTransferMarkt
        self.stade = Stade().fromJson(
            json=json['stade']) if json.get('stade') else self.stade

        return self
